gen/src/generate/char_property/bsearch_set.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::BTreeSet;
+
+/// A simple range-collapsing binary search array slice.
```

Comment:
I'd argue that it makes implementation of a contains fn harder.

In the `[(low, high)]` case, you bsearch for the containing pair and if you found it, you're in the set, and if you didn't, you're out.

In the `[low, high, ]` case, you bsearch for your value, and then have to handle two cases: found it or not. In the case that you found it, if the index is even you're in, odd, you're out. In the case that you didn't find it, the index you get is the safe insertion point, so the element higher than your value, so if the index is even you're out, odd, you're in.

I think the pairs is simpler (and consistent with the value lookup).